### PR TITLE
Remove CPU pin from GraphGather

### LINKED ISSUE #1157: Remove GraphGather pin to CPU on upgrade to TF 1.6
The bug requiring us to pin to CPU is fixed in tf 1.8

https://github.com/tensorflow/tensorflow/issues/16872#issuecomment-368016454

https://github.com/deepchem/deepchem/pull/1081

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes #1157

### CODE REVIEW

File: deepchem/models/tensorgraph/layers.py [hunk 1]
```diff
@@ -2489,7 +2489,6 @@ def __init__(self, batch_size, activation_fn=None, **kwargs):
     super(GraphGather, self).__init__(**kwargs)
 
   def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
-    with tf.device('/cpu'):
       inputs = self._get_input_tensors(in_layers)
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/layers.py]: You will have to un-indent things that were in this indentation block.

Comment (souravsingh) [reply deepchem/models/tensorgraph/layers.py]: Done

**Comment by souravsingh** [comment]
@lilleswing The patch is ready to merge

**Comment by lilleswing** [comment]
LGTM!

**Pull request merged**